Sources/HiPMobile/BusinessLayer/ContentApiFetchers/AchievementFetcher.cs

```diff
@@ -44,10 +44,23 @@ public AchievementFetcher(IAchievementsApiAccess client)
                 .Select(it => it.Id.ToString()).ToList();
 
             var achievements = AchievementConverter.Convert(achievementDtos, dataAccess).ToList();
+
             foreach (var unlocked in achievements.Where(it => unlockedAchievementIds.Contains(it.Id)))
+             {
+                 unlocked.IsUnlocked = true;
```

Comment:
I believe this should be done inside a transaction

Reply:
I did not change this part, since it modifies the achievements before adding them to the database. The modified ones and the others are then added to the database. But we can change it, when we look for the other problem.